_posts/2022-03-16-solidity-0.8.13-release-announcement.md

```diff
@@ -0,0 +1,243 @@
+from the optimized version. Without the optimizer, you are also more likely to run into "stack too deep"
+issues which can often be avoided by the optimizer as explained below.
+
+The performance of the new pipeline is not yet always superior to the old one, but it
+can do much higher-level optimization across functions, so please try it out and give us feedback!
+
+Finally, in some edge cases, code compiled via the new compiler pipeline
+behaves differently than code compiled via the old compiler pipeline.
+The most prominent is the initialization order of state variables that depend
+on the result of constructors in other contracts.
+Others are related to weird uses of modifiers, dependence of evaluation order inside
+statements or dirty data. You can find a list of all the changes
+in our [documentation](https://docs.soliditylang.org/en/develop/ir-breaking-changes.html).
+
+### Memory-Safe Assembly / Stack Too Deep
+
+One important feature of the new compiler pipeline is that the Yul optimizer can move stack variables to memory
+and thus avoid the "stack too deep" issue in a lot of cases. In order to do this safely, the compiler
+needs to know that the memory slots it intends to use are not used by inline assembly.
+
+Inline assembly is considered memory-safe if it only uses memory that has been previously allocated
+either by high-level Solidity code or by reading from the free memory pointer at ``0x40``.
+At the very beginning of the code, the compiler sets the free memory pointer to a position
+after the area reserved for the stack variables that are moved to memory.
+
+The compiler considers inline assembly blocks as memory safe only for
+very simple blocks that do not have any opcodes that access memory and also do not access
+Solidity variables related to memory reference types.
+For all other blocks, it is difficult to impossible to determine this automatically in a safe way
+and thus we added a syntactic mechanism for the developer to assert it:
+
+```
+assembly ("memory-safe") { ... }
+```
+
+Marking an unsafe block as memory-safe can lead to undefined behavior
+due to conflicts between stack variables and memory, the compiler will
+not check that marked blocks are actually memory safe.
+Please see the [documentation](https://docs.soliditylang.org/en/v0.8.13/assembly.html#memory-safety)
+for more information about when an inline assembyl block is memory safe.
+
+Inline assembly blocks that access memory are not considered memory-safe by default and
+even a single such block disables the stack-to-memory mechanism for the whole contract.
+Assuming otherwise would be dangerous and due to the risk of stack variables moved to memory being overwritten.
+
+If you are e.g. writing a code library that is meant to be used across compiler versions,
+you can also use a special comment syntax that has the same effect:
+/// @solidity memory-safe-assembly
+assembly { ... }
```

Comment:
This is missing a code block.
````suggestion
```
/// @solidity memory-safe-assembly
assembly { ... }
```
````

Now that I think of it, I probably used the triple backtick inside a `suggestion` block in my previous suggestion so this might not have been visible.